Sources/Segment/Utilities/Storage.swift

```diff
@@ -241,23 +243,20 @@ extension Storage {
         }
         
         syncQueue.sync {
-            do {
-                let jsonString = event.toString()
-                if let jsonData = jsonString.data(using: .utf8) {
-                    let handle = try FileHandle(forWritingTo: storeFile)
-                    handle.seekToEndOfFile()
-                    // prepare for the next entry
-                    if newFile == false {
-                        handle.write(",".data(using: .utf8)!)
-                    }
-                    // write the data
-                    handle.write(jsonData)
-                    handle.closeFile()
-                } else {
-                    assert(false, "Storage: Unable to convert event to json!")
+            let jsonString = event.toString()
+            if let jsonData = jsonString.data(using: .utf8) {
+                fileHandle?.seekToEndOfFile()
+                // prepare for the next entry
+                if newFile == false {
+                    fileHandle?.write(",".data(using: .utf8)!)
                 }
-            } catch {
-                assert(false, "Storage: failed to write event to \(storeFile), error: \(error)")
+                // write the data
+                fileHandle?.write(jsonData)
+                if #available(tvOS 13, *) {
```

Comment:
Might be good to add a comment why we are doing this for 13 and above but supporting 11.